IPython/utils/tests/test_tokenutil.py

```diff
@@ -29,65 +37,91 @@ def test_simple():
-    ])
-    expected = 'hello'
+@pytest.mark.parametrize(
+    "cell",
+    [
+        "\n".join(["a = 5", "b = hello('string', there)"]),
+        "\n".join(
+            [
+                '"""\n\nxxxxxxxxxx\n\n"""',
+                '5, """',
+                "docstring",
+                "multiline token",
+                '""", [',
+                '2, 3, "complicated"]',
+                'b = hello("string", there)',
+            ]
+        ),
+    ],
+)
+def test_multiline(cell):
+    expected = "hello"
     start = cell.index(expected) + 1
     for i in range(start, start + len(expected)):
         expect_token(expected, cell, i)
-    expected = 'hello'
     start = cell.index(expected) + 1
     for i in range(start, start + len(expected)):
         expect_token(expected, cell, i)
 
-def test_nested_call():
+
+def test_nested_call_kwargs():
     cell = "foo(bar(a=5), b=10)"
-    expected = 'foo'
-    start = cell.index('bar') + 1
-    for i in range(start, start + 3):
-        expect_token(expected, cell, i)
-    expected = 'bar'
-    start = cell.index('a=')
+    start = cell.index("bar")
+    last = start + len("bar")
+    for i in range(start, last + 1):
+        expect_token("bar", cell, i)
+    start = cell.index("a=")
     for i in range(start, start + 3):
-        expect_token(expected, cell, i)
-    expected = 'foo'
-    start = cell.index(')') + 1
-    for i in range(start, len(cell)-1):
-        expect_token(expected, cell, i)
+        expect_token("bar", cell, i)
+    expect_token("bar", "foo(bar(a=|5), b=10)")
+    expect_token("bar", "foo(bar(a=5|), b=10)")
+    for i in range(cell.index(")") + 1, cell.index("b=") + 2):
+        expect_token("foo", cell, i)
+    expect_token("foo", cell, len(cell) - 1)
+
+
+def test_nested_call_args():
+    expect_token("bar", "foo(bar(x,| ))")
+    expect_token("bar", "foo(bar(x, |))")
+    expect_token("bar", "foo(ba|r(x, y))")
+    expect_token("x", "foo(bar(|x, y))")
+    expect_token("bar", "foo(bar(x|, y))")
+    expect_token("bar", "foo(bar(x,| y))")
+    expect_token("y", "foo(bar(x, |y))")
+    expect_token("bar", "foo(bar(x, y|))")
+    expect_token("foo", "foo(bar(x, y)|)")
+    expect_token("foo", "foo(bar(, y)|)")
+
+
+def test_outer_name():
+    expect_token("x", "|x + foo(a, b) + y + z")
+    expect_token("x", "x| + foo(a, b) + y + z")
+    expect_token("x", "x |+ foo(a, b) + y + z")
+    expect_token("x", "x +| foo(a, b) + y + z")
+    expect_token("foo", "x + |foo(a, b) + y + z")
+    expect_token("a", "x + foo(|a, b) + y + z")
+    expect_token("foo", "x + foo(a|, b) + y + z")
+    expect_token("foo", "x + foo(a, b)| + y + z")
+    expect_token("foo", "x + foo(a, b) |+ y + z")
+    expect_token("foo", "x + foo(a, b) +| y + z")
+    expect_token("y", "x + foo(a, b) + |y + z")
+    expect_token("y", "x + foo(a, b) + y| + z")
+    expect_token("y", "x + foo(a, b) + y |+ z")
+    expect_token("y", "x + foo(a, b) + y +| z")
+    expect_token("z", "x + foo(a, b) + y + |z")
+
```

Comment:
probably coudl make those @pytest.mark.parametrize as well.

Reply:
True, but I think it's more readable as-is.